api/src/main/java/org/openmrs/module/patientqueueing/QueueingUtils.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * <p>
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.patientqueueing;
+
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class QueueingUtils {
+	
+	private QueueingUtils() {
+	}
+	
+	/**
+	 * This method changes the time of a date set to the time given in the time string.
+	 * The Time format should be HH:mm:ss forexample 00:00:00 ,23:59:59 etc
+	 * 
+	 * @param date This is the date to be converted into set date format
+	 * @param time The time String which is appended to the date in-case it is not null
+	 * @return Date which is converted to required format
+	 * @throws ParseException
+	 */
+	public static Date changeTimeOfDate(Date date, String time) throws ParseException {
+		AdministrationService administrationService = Context.getAdministrationService();
+		SimpleDateFormat formatter = new SimpleDateFormat(administrationService.getGlobalProperty("patientqueueing.displayDateTimeFormat"));
+		String formattedDate;
+		SimpleDateFormat formatterExt = new SimpleDateFormat(administrationService.getGlobalProperty("patientqueueing.defaultDateFormat"));
+		if (time != null) {
+			formattedDate = formatterExt.format(date) + " " + time;
+		} else {
+			formattedDate = formatterExt.format(date);
+		}
+		
+		return formatter.parse(formattedDate);
+	}
+	
+	/**
+	 * This Method converts date to a string date format.
+	 * 
+	 * @param date This is the date to be converted into set date format and returns a string of
+	 *            that format
+	 * @param time The time String which is appended to the date in-case it is not null
+	 * @return String date which is converted to required format
+	 */
+	public static String formatDateAsString(Date date, String time) {
```

Comment:
Why would you pass in time, when you are simply formatting a date as a string? Doesn't the date already have time included?

Reply:
@dkayiwa it does but Sometimes you may need to modify the time of a date so It serves the purpose as well.

Reply:
What is the current use case for that?

Reply:
Kindly Check the generateQueueNumber method in the PatientQueueingServiceImpl atQueueingUtils.formatDateAsString(date, null);